beacon_node/http_api/src/lib.rs

```diff
@@ -1631,49 +1712,63 @@ pub fn serve<T: BeaconChainTypes>(
         .and(network_tx_filter.clone())
         .and_then(
             |chain: Arc<BeaconChain<T>>,
-             aggregate: SignedAggregateAndProof<T::EthSpec>,
+             aggregates: Vec<SignedAggregateAndProof<T::EthSpec>>,
              network_tx: UnboundedSender<NetworkMessage<T::EthSpec>>| {
                 blocking_json_task(move || {
-                    let aggregate =
+                    let mut verified_aggregates = Vec::new();
```

Comment:
We could avoid some re-allocations here if we use `Vec::with_capacity(aggregates.len())` here.

We might end up allocating a bit extra in the case of an error, but I think it's fine to optimize for the success path here.